### PR TITLE
chore: tests and module update from template

### PR DESCRIPTION
**Pull request by SpecialistSun**
- chore: copied missing files from template
- fix: fix broken links

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -8,20 +8,9 @@ variable "cloud_function" {
...
-variable "region" {
-  description = "(Optional) The location of this cloud function. Used to find the parent resource to bind the IAM policy to. If not specified, the value will be parsed from the identifier of the parent resource. If no region is provided in the parent identifier and no region is specified, it is taken from the provider configuration."
-  type        = string
-  default     = null
-}
-
 variable "members" {
   type        = set(string)
-  description = "(Optional) Identities that will be granted the privilege in role. Each entry can have one of the following values: 'allUsers', 'allAuthenticatedUsers', 'user:{emailid}', 'serviceAccount:{emailid}', 'group:{emailid}', 'domain:{domain}', 'projectOwner:projectid', 'projectEditor:projectid', 'projectViewer:projectid'."
+  description = "(Required) Identities that will be granted the privilege in role. Each entry can have one of the following values: 'allUsers', 'allAuthenticatedUsers', 'user:{emailid}', 'serviceAccount:{emailid}', 'group:{emailid}', 'domain:{domain}', 'projectOwner:projectid', 'projectEditor:projectid', 'projectViewer:projectid'."
```
Comment (mariux) [inline variables.tf:13]: members is not required by the module. please revert the change. (it defaults to empty list of members)

File: variables.tf [hunk 2]
```diff
@@ -31,7 +20,18 @@ variable "members" {
 }
 
 variable "role" {
-  description = "(Optional) The role that should be applied. Only one 'iam_binding' can be used per role. Note that custom roles must be of the format '[projects|organizations]/{parent-name}/roles/{role-name}'."
+  description = "(Required) The role that should be applied. Only one 'iam_binding' can be used per role. Note that custom roles must be of the format '[projects|organizations]/{parent-name}/roles/{role-name}'."
```
Comment (mariux) [inline variables.tf:23]: role is not required by the module. please revert the change. in case policy bindings are used, role does not need to be set.

**Review by mariux — CHANGES_REQUESTED** [review]
Please revert changes in variables.tf.

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -4,10 +4,11 @@ on:
   push:
     branches:
       - main
+      - master
   pull_request:
 
 concurrency:
-  group: terraform-google-cloud-function-iam
+  group: terraform-module-template
```
Comment (mariux) [inline .github/workflows/tests.yml:11]: please run `make template/adjust` to resolve the name change.

**Review by mariux — CHANGES_REQUESTED** [review]

File: .github/workflows/tests.yml [hunk 4]
```diff
@@ -16,7 +17,84 @@ jobs:
...
+      - name: Run Unit Tests
+        if: steps.changes.outputs.terraform == 'true'
+        run: make test/unit-tests
+        env:
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.MINEIROS_TESTING_AWS_SECRET_ACCESS_KEY }}
+          AWS_ACCESS_KEY_ID: ${{ secrets.MINEIROS_TESTING_AWS_ACCESS_KEY_ID }}
+          GOOGLE_CREDENTIALS: ${{ secrets.MINEIROS_TESTING_GCP_SA_KEY_FILE }}
+          TEST_GCP_PROJECT: ${{ secrets.MINEIROS_TESTING_GCP_PROJECT }}
+          TEST_GCP_ORG_DOMAIN: ${{ secrets.MINEIROS_TESTING_GCP_ORG_DOMAIN }}
+          TEST_GCP_BILLING_ACCOUNT: ${{ secrets.MINEIROS_TESTING_GCP_BILLING_ACCOUNT }}
```
Comment (mariux) [inline .github/workflows/tests.yml:100]: Please adjust editor settings to always include EOL at EOF.

**Review by mariux — CHANGES_REQUESTED** [review]

File: test/terramate_provider_versions.tm.hcl [hunk 5]
```diff
@@ -0,0 +1,5 @@
+globals {
+  provider                    = "google"
+  minimum_provider_version    = "4.0"
+  provider_version_constraint = "~> ${global.minimum_provider_version}"
+}
```
Comment (mariux) [inline test/terramate_provider_versions.tm.hcl:5]: this file is not part of the template repository. please use versions.tm.hcl from template repository.

**Review by mariux — CHANGES_REQUESTED** [review]

File: .pre-commit-config.yaml [hunk 6]
```diff
@@ -1,13 +1,13 @@
...
       - id: terraform-fmt
       - id: terraform-validate
         exclude: ^examples|.terraform/
-      - id: tflint
-      - id: golangci-lint
-      - id: phony-targets
-      - id: markdown-link-check
-        args: ['-p']   # When adding the -p flag, markdown-link-check will always with an exit code 0, even if dead links are found
-        verbose: true  # Forces the output of the hook to be printed even when the hook passes.
+      - id: tflint
```
Comment (zied-elouaer) [inline .pre-commit-config.yaml:13]: @SpecialistSun please add new lines at the end of each file. You can configure your IDE to add them automatically to make it easier.

File: test/unit-minimal/main.tf [hunk 7]
```diff
@@ -0,0 +1,54 @@
+module "test" {
+  source = "../.."
+
+  # add all required arguments
+
+  cloud_function = "test"
```
Comment (zied-elouaer) [inline test/unit-minimal/main.tf:6]: Please add only required arguments in the minimal test. Basically anything that does not have a default value.

**Review by zied-elouaer — CHANGES_REQUESTED** [review]
Please check my comments. Otherwise looks good so you can go ahead and start adding actual unit tests. Thanks!

File: .pre-commit-config.yaml (hunk 6 above)
Comment (switchdk) [reply .pre-commit-config.yaml]: @zied-elouaer just curious about adding new lines. Since it seems something rather easy to forget and difficult to know, could we use pre-commit to manage it? At least that is what I have done before to avoid having to configure particular settings.

Comment (zied-elouaer) [reply .pre-commit-config.yaml]: @switchdk that sounds like a good idea. Maybe you should do it :smile:

File: test/unit-minimal/main.tf [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+
+  # add all optional arguments that create additional/extended resources
+
+  role = "roles/viewer"
+
+  members = ["user:serdar.yildirim@mineiros.io"]
```
Comment (mariux) [inline test/unit-minimal/main.tf:12]: please do not use real users here.

**Review by mariux — CHANGES_REQUESTED** [review]
tests are mostly empty and missing. Please add the missing code for the tests in a next PR.

requested changes: 
- please replace real email address with example one.
- please remove dead code that is commented in the tests or replace it with actual test code (this or next PR)

**Review by mariux — CHANGES_REQUESTED** [review]
why is this marked as a breaking change? please change ;)

**Review by mariux — APPROVED** [review]
lgtm
